### DISCUSSION TITLE
GraphQLFieldResolver with TArgs does not work with GraphQLResolverMap

### DISCUSSION THREAD

**Discussion by eliw00d** [apollographql/apollo-server#4982] [Q&A]
I tried to split my resolver code up into smaller chunks as our resolvers file was getting large. We use TypeScript and I wanted to type each resolver using `GraphQLFieldResolver`. I provided `TSource`, `TContext`, and `TArgs` but noticed I was getting errors related to `TArgs`. Because `GraphQLResolverMap` does not provide anything for `TArgs`, it will always use the default `TArgs`. 

```typescript
export interface GraphQLResolverMap<TContext = {}> {
  [typeName: string]:
    | {
        [fieldName: string]:
          | GraphQLFieldResolver<any, TContext> // missing TArgs
          | {
              requires?: string;
              resolve: GraphQLFieldResolver<any, TContext>; // missing TArgs
            };
      }
    | GraphQLScalarType
    | {
        [enumValue: string]: string | number;
      };
}
```

```typescript
export type GraphQLFieldResolver<
  TSource,
  TContext,
  TArgs = { [argName: string]: any } // default supplied here
> = (
  source: TSource,
  args: TArgs,
  context: TContext,
  info: GraphQLResolveInfo,
) => any;
```

Even if I did not split my resolvers up, we would have to use the default `TArgs` due to how `GraphQLResolverMap` is built. 

Is this intended or are there plans to support this down the road?

**Comment by glasser** [comment]
Are you writing your types by hand, or doing something like [GraphQL Code Generator's TypeScript Resolvers support](https://graphql-code-generator.com/docs/plugins/typescript-resolvers#integration-with-apollo-server) ?

**Reply by eliw00d** [reply]
Currently by hand.

**Reply by glasser** [reply]
You might be happier using generated types, or at least playing with GCG to see how its generated types solve this problem (make sure to use the option documented there as necessary for Apollo Server compatibility).

**Reply by eliw00d** [reply]
Alright, I'll take a look. Thanks!
